### DISCUSSION TITLE
autostart application in .config/autostart/

### DISCUSSION THREAD

**Discussion by SimoneRiccardi** [hyprwm/Hyprland#3389] [General]
XDG defines the folder "$XDG_CONFIG_DIRS/autostart" for starting applications at the beginning of a session. It would be a good alternative to exec-once for applications that have autostart option in their settings

**Comment by memchr** [comment]
Doesn't systemd already provide a systemd-xdg-autostart-generator?

**Reply by SimoneRiccardi** [reply]
As I've readed form [systemd](https://systemd.io/DESKTOP_ENVIRONMENTS/#xdg-autostart-integration) and [freedesktop](https://www.freedesktop.org/software/systemd/man/systemd-xdg-autostart-generator.html) docs systemd-xdg-autostart-generator should be used by the desktop enviroment to delegate the autostart to systemd. I've not been capable to use it by my own.

But there are some [alternatives](https://wiki.archlinux.org/title/XDG_Autostart).

**Reply by agaskins** [reply]
it doesn't seem to be working for me. I can manually run an xdg generated autostart, for example `systemctl --user start app-blueman@autostart`, and it pops right up in the bar. However, my understanding is that these should start when a graphical environment is started, right?

Maybe I am misunderstanding how this works (very likely).

**Reply by memchr** [reply]
Make sure the `xdg-desktop-autostart.target` is included in your desktop session, as the generated autostart units will be installed into this target.

Note that this target cannot be started directly with `systemctl --user start xdg-desktop-autostart.target`. Instead, you must create a new target for your desktop session and use the `Wants=` directive to include `xdg-desktop-autostart.target`.

For example, create a file called `hyprland-session.target` in your `~/.config/systemd/user/` directory with the following content

```ini
[Unit]
Description=Hyprland session target
BindsTo=graphical-session.target
Wants=xdg-desktop-autostart.target
```

Then add `exec-once = systemctl --user start hyprland-session.target` to your hyprland configuration file.
The xdg autostarts will work the next time you log in.

**Reply by 696558586537** [reply]
Lebanon
